### PR TITLE
Pandas 2.2.0

### PR DESCRIPTION
**Pull request by AdamOrmondroyd**
# Description

Pandas 2.2.0 just released, with a flurry of changes to the plotting to become more stateless (though not even close to achieving it), and a whole flood of deprecations ready for pandas 3.0 (!!!!!!!)

Unfortunately, the changes to the plotting are incompatible with the old

I'm yet to do anything about the warnings

Fixes # (issue)

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code is PEP8 compliant (`flake8 anesthetic tests`)
- [x] My code contains compliant docstrings (`pydocstyle --convention=numpy anesthetic`)
- [x] New and existing unit tests pass locally with my changes (`python -m pytest`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have appropriately incremented the [semantic version number](https://semver.org/) in both README.rst and anesthetic/_version.py

### CODE REVIEW

File: anesthetic/plotting/_matplotlib/hist.py [hunk 1]
```diff
@@ -180,18 +179,18 @@ def __init__(
     ) -> None:
         super().__init__(data, bins=bins, bottom=bottom, **kwargs)
 
-    def _calculate_bins(self, data):
-        if 'range' not in self.kwds or self.kwds['range'] is None:
+    def _calculate_bins(self, data, bins):
+        nd_values = data.infer_objects(copy=False)._get_numeric_data()
+        values = np.ravel(nd_values)
+        values = values[~isna(values)]
```
Comment (AdamOrmondroyd) [inline anesthetic/plotting/_matplotlib/hist.py:185]: I'm not happy with repeating this a third time, and I haven't checked if the weights definitely need checking for nans

**Comment by AdamOrmondroyd** [comment]
I think that's all the new deprecation warnings, aside from the warnings about constructing from a `BlockManager`. Discussion on this seems to be ongoing, so I'm tempted to wait until pandas 2.2.1 to see if I still need to deal with these. https://github.com/pandas-dev/pandas/issues/57032

**Comment by AdamOrmondroyd** [comment]
PR here which may clean up some of the BlockManager mess, https://github.com/pandas-dev/pandas/pull/57553

File: anesthetic/plotting/_matplotlib/hist.py (hunk 1 above)
Comment (lukashergt) [reply anesthetic/plotting/_matplotlib/hist.py:182]: Well, considering that `Hist1dPlot` was created for a `WeightedSeries`, we could simplify things here. 
Even simpler than my previous suggestion:
```python
            xmin, xmax = data.quantile(q).to_numpy().ravel()
```

Not sure what is easier for maintenance in the long run, though, so happy with either approach.

**Review by lukashergt — DISMISSED** [review]
Is there anything else that needs to be taken into account for this?

If not, I am happy with this. Thanks for catching this so quickly @AdamOrmondroyd!

Feel free to handle my inline reply as you see fit.

File: anesthetic/plotting/_matplotlib/hist.py (hunk 1 above)
Comment (AdamOrmondroyd) [reply anesthetic/plotting/_matplotlib/hist.py:182]: That's too tidy to resist!

**Review by lukashergt — APPROVED** [review]
Thanks @AdamOrmondroyd!

**Pull request merged**